lib/galaxy/tools/verify/interactor.py

```diff
@@ -341,9 +341,9 @@ def stage_data_async(self, test_data, history_id, tool_id, force_path_paste=Fals
                 tool_input.update({
                     "files_%d|type" % i: "upload_dataset",
                 })
-            name = test_data['name']
+            name = os.path.basename(test_data['name'])
```

Comment:
The change on line 346 should be enough for https://github.com/galaxyproject/tools-iuc/pull/2439 , are you that's actually a path?

Reply:
Not sure about anything .. as usual. My coding in Galaxy is still often based on intuition :/ 

The change on 364 applies `basename` to `fname` which should be equal to `test_data['fname']` (https://github.com/galaxyproject/galaxy/blob/da91d724bcd4329098af01d5807c25bcfa364d85/lib/galaxy/tools/verify/interactor.py#L318). So I intuitively applied basename in both branches of the if statement. Initially I thought about applying directly on line 318. But fname is used at multiple other places, e.g. https://github.com/galaxyproject/galaxy/blob/da91d724bcd4329098af01d5807c25bcfa364d85/lib/galaxy/tools/verify/interactor.py#L353

Reply:
I agree with the change on line 346, but I think the one here may be not needed, since that's on `test_data['name']`, not `test_data['fname']` .